source/extensions/omni.isaac.lab/test/assets/test_rigid_object.py

```diff
@@ -118,6 +168,44 @@ def test_initialization(self):
                             # update object
                             cube_object.update(sim.cfg.dt)
 
+    def test_initialization_with_heterogeneous_cubes(self):
+        """Test initialization for heterogeneous prims with rigid body API at different prim paths."""
+        num_cubes = 2
+        for device in ("cuda:0", "cpu"):
+            with self.subTest(num_cubes=num_cubes, device=device):
+                with build_simulation_context(device=device, auto_add_lighting=True) as sim:
+                    # Generate cubes configs
+                    cube_object_cfg, spawn_cfgs, _ = generate_heterogeneous_cubes_scene(
+                        num_cubes=num_cubes, device=device
+                    )
+
+                    # Spawn objects to scene
+                    for cfg in spawn_cfgs:
+                        RigidObject(cfg)
```

Comment:
Why is there a spawn again if the function above generates the heterogenous scene?

Reply:
moved spawning to the function